prow/crier/reporters/slack/reporter.go

```diff
@@ -75,11 +80,16 @@ func channel(prowCfg config.SlackReporter, jobCfg *v1.SlackReporterConfig) (stri
 	return host, channel
```

Comment:
I think we should either emit an error is `channel` is not specified or set it to a default value like `host` if there is a reasonable default.

Reply:
I don't think that would happen. Added comments on `ShouldReport` below, which returns true only if:
- global config is defined
- (OR) slack config in job is defined and slack channel is defined in job

The first case isn't covered by this PR, but not covered currently either
The second case wouldn't result in empty channel